saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -705,22 +705,54 @@ def perform_mutation(cls, root, info, **data):
             warehouses = cls.get_nodes_or_error(
                 warehouse_ids, "warehouse", only_type=Warehouse
             )
-            cls.update_or_create_variant_stocks(variant, stocks, warehouses)
+            cls.update_or_create_variant_stocks(variant, stocks, warehouses, info)
 
         variant = ChannelContext(node=variant, channel_slug=None)
         return cls(product_variant=variant)
 
     @classmethod
     @traced_atomic_transaction()
-    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses):
+    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses, info):
         stocks = []
         for stock_data, warehouse in zip(stocks_data, warehouses):
-            stock, _ = warehouse_models.Stock.objects.get_or_create(
+            stock, created = warehouse_models.Stock.objects.get_or_create(
                 product_variant=variant, warehouse=warehouse
             )
+            stocks.append(
+                {
+                    "stock": stock,
+                    "created": created,
+                    "prev_quantity": stock.quantity,
+                    "current_quantity": stock_data["quantity"],
+                }
+            )
             stock.quantity = stock_data["quantity"]
-            stocks.append(stock)
-        warehouse_models.Stock.objects.bulk_update(stocks, ["quantity"])
+        stocks_to_update = [s["stock"] for s in stocks]
```

Comment:
- vars should have meaningfull names, avoid using short ones like `s`.
- even if we name it `stock`, this line would be `[stock["stock"] for stock in stocks]` which is confusing
- `stocks` array should be renamed as it is no longer an array of stock objects, I would name it `stocks_data` and rename the incoming fuction arg to `stocks` to keep signature consistent.